src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,30 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector) {
+    super(popupSelector);
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._saveButton = this._popupForm.querySelector(".modal__button");
+    this._saveButtonText = this._saveButton.textContent;
+  }
+
+  setLoading(isLoading) {
```

Comment:
You've implemented the logic for rendering loading via the class methods, Excellent!